examples/text_to_image/train_text_to_image.py

```diff
@@ -541,7 +614,8 @@ def collate_fn(examples):
     unet, optimizer, train_dataloader, lr_scheduler = accelerator.prepare(
         unet, optimizer, train_dataloader, lr_scheduler
     )
-    accelerator.register_for_checkpointing(lr_scheduler)
```

Comment:
It's not required to register `lr_scheduler` here, it's automatically checkpointed by `accelerate`. We only need to register custom objects.

Reply:
Interesting, thanks. [This documentation](https://huggingface.co/docs/accelerate/usage_guides/checkpoint) led me to believe we needed to register it, but in those examples the learning rate scheduler is not being passed to `prepare`.

Reply:
Hmm should we maybe ask on the `accelerate` repo?

Reply:
I've verified this, all standard objects that we pass to `prepare` (like `nn.Module`, `DataLoader`, `Optimizer`, `Scheduler`) are automatically checkpointed by `accelerate`. We only need to register custom objects or models that we don't pass to `prepare`.